io/ntriples/common/src/main/scala/org/w3/banana/io/NTriplesWriter.scala

```diff
@@ -24,26 +27,35 @@ class NTriplesWriter[Rdf <: RDF](implicit val ops:RDFOps[Rdf]) extends RDFWriter
    * @param node Rdf node
    * @return
    */
-  def node2Str(node:Rdf#Node): String = ops.foldNode(node)(
-    { case ops.URI(url) => "<" + url + ">"},
-    { case ops.BNode(id) => "_:" + id},
-    {
-      case ops.Literal((string, ops.URI(datatype), Some(lang))) => "\"" + string + "\"" + "@" + lang
-      case ops.Literal((string, ops.URI(datatype), None)) => "\"" + string + "\"" + "^^<" + datatype + ">"
+  def node2Str(node: Rdf#Node, base: Rdf#URI): String = ops.foldNode(node)(
+    uri => "<" + asString(uri, base) + ">",
```

Comment:
But the tests don't test it - I Know 'cos I've added some `println`'s and none of the tests actually add the base, so you you need some tests with some URI's that need resolving - eg 
```
Triple(URI("betehess"), foaf.homepage, URI("http://bertails.org/"))
```